From 339f603ab6b90df65de5d517ae2f692c35a9cd99 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Mon, 6 Feb 2023 17:28:49 +0100
Subject: [PATCH 1/1] meson.build: enable building on riscv64

The meson specific files up to now lack support for the RISC-V
architecture.

Part-of: <https://gitlab.freedesktop.org/gstreamer/meson-ports/libffi/-/merge_requests/18>
Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Origin: https://gitlab.freedesktop.org/gstreamer/meson-ports/libffi/-/commit/339f603ab6b90df65de5d517ae2f692c35a9cd99
---
 include/ffi_noarch.h.meson | 2 ++
 meson.build                | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/include/ffi_noarch.h.meson b/include/ffi_noarch.h.meson
index 23bcd9c..5662eef 100644
--- a/include/ffi_noarch.h.meson
+++ b/include/ffi_noarch.h.meson
@@ -16,6 +16,8 @@
 #include "@HEADER@-powerpc.h"
 #elif defined(__powerpc64__)
 #include "@HEADER@-powerpc64.h"
+#elif defined (__riscv) && __riscv_xlen == 64
+#include "@HEADER@-riscv64.h"
 #else
 #error "Unsupported Architecture"
 #endif
diff --git a/meson.build b/meson.build
index 7aa5fa6..ae7d0aa 100644
--- a/meson.build
+++ b/meson.build
@@ -248,6 +248,11 @@ elif host_cpu_family == 'arm'
   TARGET = 'ARM'
   c_sources = ['ffi.c']
   asm_sources = ['sysv.S']
+elif host_cpu_family == 'riscv64'
+  arch_subdir = 'riscv'
+  TARGET = 'RISCV'
+  c_sources = ['ffi.c']
+  asm_sources = ['sysv.S']
 endif
 
 if TARGET == ''
-- 
2.39.2

